### PR TITLE
Man: mention $$ as subst escape

### PR DESCRIPTION
**Pull request by mwichmann**
Some rewordings elsewhere in Variable Substitution section - mainly to a variable that's a function.

This is a doc-only change.  There's a larger change to this section to be proposed, but submitting this one first to keep things a little simpler.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: doc/man/scons.xml [hunk 1]
```diff
@@ -6414,18 +6414,25 @@ env.Command('marker', 'input_file', action=[MyBuildAction, Touch('$TARGET')])
 &scons;
 performs variable substitution on the string that makes up
 the action part of the builder.
-Variables to be interpolated are indicated in the
-string with a leading
-<literal>$</literal>, to distinguish them from plain text
+Variables or other text to be substituted are indicated in the
```
Comment (bdbaddog) [inline doc/man/scons.xml]: When would other text work? $OTHER_TEXT when not a variable is just blank..

Comment (mwichmann) [reply doc/man/scons.xml]: the planned followon will improve on the way this is structured... things like a python expression are not variables, thus the "other"

Comment (bdbaddog) [reply doc/man/scons.xml]: but python expressions are only valid inside ${ }, and not $xyz

Comment (mwichmann) [reply doc/man/scons.xml]: Yes, that's what I meant by a restructure in a future revision.  I plan to propose it as a substitutable is `${text}` but under these circumstances you can leave out the `{}`, which I think makes it more consistent,  I can pull out the "or other text" from here for now if that makes you happier, don't know when I'll get to finishing up the next round.

**Comment by mwichmann** [comment]
Anything I need to do for this one?  As noticed there's a followon changeset that restructures the section a bit.

**Comment by bdbaddog** [comment]
I keep thinking it'd be good to list this in a table or some such with the other $ expansions rather than just in the paragraph..
Thoughts?

**Comment by mwichmann** [comment]
> I keep thinking it'd be good to list this in a table or some such with the other $ expansions rather than just in the paragraph.. Thoughts?

It's in a list in the next change.  Unfortunately something has messed up on this end so I can't rebase things easily, but a build of the unpushed branch has this:

---

SCons uses the following rules when converting construction variables into command line strings:

* If the value is a string it is interpreted as space delimited command line arguments.
* If the value is a list it is interpreted as a list of command line arguments. Each element of the list is converted to a string.
* Anything that is not a list or string is converted to a string and interpreted as a single command line argument.
* Newline characters (`\n`) delimit lines. The newline parsing is done after all other parsing, so it is not possible for arguments (e.g. file names) to contain embedded newline characters.
* Substitution can be suppressed be prepending an additional `$`. For example, `$$FOO` will be left in the final string as `$FOO`.

**Comment by bdbaddog** [comment]
> > I keep thinking it'd be good to list this in a table or some such with the other $ expansions rather than just in the paragraph.. Thoughts?
> 
> It's in a list in the next change. Unfortunately something has messed up on this end so I can't rebase things easily, but a build of the unpushed branch has this:
> 
> SCons uses the following rules when converting construction variables into command line strings:
> 
> * If the value is a string it is interpreted as space delimited command line arguments.
> * If the value is a list it is interpreted as a list of command line arguments. Each element of the list is converted to a string.
> * Anything that is not a list or string is converted to a string and interpreted as a single command line argument.
> * Newline characters (`\n`) delimit lines. The newline parsing is done after all other parsing, so it is not possible for arguments (e.g. file names) to contain embedded newline characters.
> * Substitution can be suppressed be prepending an additional `$`. For example, `$$FOO` will be left in the final string as `$FOO`.

Maybe for a literal '$' use '$$' ?

**Comment by mwichmann** [comment]
> Maybe for a literal '$' use '$$' ?

I like that - updating the branch here.  Now to see if I can reconcile somehow....

**Pull request merged**